Terminal.Gui/Views/TabView.cs

```diff
@@ -0,0 +1,794 @@
+
+			}
+
+			/// <summary>
+			/// Renders the line with the tab names in it
+			/// </summary>
+			/// <param name="tabLocations"></param>
+			/// <param name="width"></param>
+			private void RenderTabLine (TabToRender [] tabLocations, int width)
+			{
+				int y;
+
+				if (host.Style.TabsOnBottom) {
+
+					y = 1;
+				} else {
+					y = host.Style.ShowHeaderOverline ? 1 : 0;
+				}
+
+
+
+				// clear any old text
+				Move (0, y);
+				Driver.AddStr (new string (' ', width));
+
+				foreach (var toRender in tabLocations) {
+
+					if (toRender.IsSelected) {
+						Move (toRender.X - 1, y);
+						Driver.AddRune (Driver.VLine);
+					}
+
+					Move (toRender.X, y);
+
+					// if tab is the selected one and focus is inside this control
+					if (toRender.IsSelected && host.HasFocus) {
+
+						if (host.Focused == this) {
+
+							// if focus is the tab bar ourself then show that they can switch tabs
+							Driver.SetAttribute (ColorScheme.HotFocus);
+
+						} else {
+
+							// Focus is inside the tab
+							Driver.SetAttribute (ColorScheme.HotNormal);
+						}
+					}
+
+
+					Driver.AddStr (toRender.TextToRender);
+					Driver.SetAttribute (ColorScheme.Normal);
+
+					if (toRender.IsSelected) {
+						Driver.AddRune (Driver.VLine);
+					}
+				}
+			}
+
+			/// <summary>
+			/// Renders the line of the tab that adjoins the content of the tab
+			/// </summary>
+			/// <param name="tabLocations"></param>
+			/// <param name="width"></param>
+			private void RenderUnderline (TabToRender [] tabLocations, int width)
+			{
+
+				int y;
+
+				if (host.Style.TabsOnBottom) {
+
+					y = 0;
+				} else {
+
+					y = host.Style.ShowHeaderOverline ? 2 : 1;
+				}
+
+				Move (0, y);
+
+				// If host has no border then we need to draw the solid line first (then we draw gaps over the top)
+				if (!host.Style.ShowBorder) {
+
+					for (int x = 0; x < width; x++) {
+						Driver.AddRune (Driver.HLine);
+					}
+
+
+				}
+				var selected = tabLocations.FirstOrDefault (t => t.IsSelected);
+
+				if (selected == null) {
+					return;
+				}
+
+				Move (selected.X - 1, y);
+
+				Driver.AddRune (selected.X == 1 ? Driver.VLine :
+			(host.Style.TabsOnBottom ? Driver.URCorner : Driver.LRCorner));
+
+				Driver.AddStr (new string (' ', selected.Width));
+
+
+				Driver.AddRune (selected.X + selected.Width == width - 1 ?
+		     Driver.VLine :
+				(host.Style.TabsOnBottom ? Driver.ULCorner : Driver.LLCorner));
+
+
+				// draw scroll indicators
+
+				// if there are more tabs to the left not visible
+				if (host.TabScrollOffset > 0) {
+					Move (0, y);
+
+					// indicate that
+					Driver.AddRune (Driver.LeftArrow);
+				}
+
+				// if there are mmore tabs to the right not visible
+				if (tabLocations.LastOrDefault ()?.Tab != host.Tabs.LastOrDefault ()) {
+					Move (width - 1, y);
+
+					// indicate that
+					Driver.AddRune (Driver.RightArrow);
+				}
+			}
+
+			public override bool MouseEvent (MouseEvent me)
```

Comment:
I can't get the left/right arrows to scroll tabs in the `Tab View` sample with mouse (in cmd.exe, which should work).